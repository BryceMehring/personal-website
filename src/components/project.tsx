import { Link } from './link';
import { GatsbyImage } from 'gatsby-plugin-image';

export const Project = ({
  id,
  image,
  title,
  shortTitle,
  position,
  description,
}: ProjectProp): JSX.Element => {
  const link = `/projects/${id}`;
  return (
    <div className="col-sm-12 col-md-4 col-xl-3 p-3 m-3 border shadow rounded project">
      <div className="text-center">
        <h3>
          <Link href={link}>{shortTitle || title}</Link>
        </h3>
        <p>{description}</p>
        <p>
          <b>Position</b>: {position}
        </p>
      </div>
      <div className="thumbnail">
        <Link href={link}>
          <GatsbyImage
            image={image.source.childImageSharp.gatsbyImageData}
            alt={image.alt}
            className="image"
            loading="eager"
          />
        </Link>
      </div>
    </div>
  );
};
