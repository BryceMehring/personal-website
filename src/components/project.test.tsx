import renderer from 'react-test-renderer';
import { Project } from './project';

describe('project', () => {
  it('matches snapshot', () => {
    const image: any = {
      source: {
        childImageSharp: {
          gatsbyImageData: {
            layout: 'constrained',
            backgroundColor: '#080808',
            images: {
              fallback: {
                src: '/static/e82c1/hexel.jpg',
                srcSet:
                  '/static/8710f/hexel.jpg 128w,\n/static/0ee4a/hexel.jpg 256w,\n/static/e82c1/hexel.jpg 512w',
                sizes: '(min-width: 512px) 512px, 100vw',
              },
              sources: [
                {
                  srcSet:
                    '/static/6efab/hexel.webp 128w,\n/static/20b45/hexel.webp 256w,\n/static/5999d/hexel.webp 512w',
                  type: 'image/webp',
                  sizes: '(min-width: 512px) 512px, 100vw',
                },
              ],
            },
            width: 512,
            height: 384,
          },
        },
      },
      alt: 'alt',
    };
    const wrapper = renderer.create(
      <Project
        id="test"
        title="name"
        description="short"
        position="pos"
        image={image}
      />
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
