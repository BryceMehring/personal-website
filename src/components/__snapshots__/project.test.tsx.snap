// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`project matches snapshot 1`] = `
<div
  className="col-sm-12 col-md-4 col-xl-3 card text-bg-light p-3 m-3 shadow text-center project"
>
  <div
    className="thumbnail"
  >
    <a
      href="/projects/test"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="gatsby-image-wrapper gatsby-image-wrapper-constrained"
        data-gatsby-image-wrapper=""
        style={
          {
            "backgroundColor": undefined,
            "display": "inline-block",
            "overflow": "hidden",
            "position": "relative",
            "verticalAlign": "top",
          }
        }
      >
        <div
          style={
            {
              "display": "block",
              "maxWidth": 512,
            }
          }
        >
          <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="data:image/svg+xml;charset=utf-8,%3Csvg%20height='384'%20width='512'%20xmlns='http://www.w3.org/2000/svg'%20version='1.1'%3E%3C/svg%3E"
            style={
              {
                "display": "block",
                "maxWidth": "100%",
                "position": "static",
              }
            }
          />
        </div>
        <div
          aria-hidden={true}
          data-placeholder-image=""
          style={
            {
              "height": "100%",
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": "100%",
            }
          }
        />
        <picture>
          <source
            sizes="(min-width: 512px) 512px, 100vw"
            srcSet="/static/6efab/hexel.webp 128w,/static/20b45/hexel.webp 256w,/static/5999d/hexel.webp 512w"
            type="image/webp"
          />
          <img
            alt="alt"
            data-gatsby-image-ssr=""
            data-main-image=""
            decoding="async"
            loading="eager"
            sizes="(min-width: 512px) 512px, 100vw"
            src="/static/e82c1/hexel.jpg"
            srcSet="/static/8710f/hexel.jpg 128w,/static/0ee4a/hexel.jpg 256w,/static/e82c1/hexel.jpg 512w"
            style={
              {
                "backgroundColor": undefined,
                "height": "100%",
                "left": 0,
                "objectFit": undefined,
                "objectPosition": undefined,
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transform": "translateZ(0)",
                "transition": "opacity 250ms linear",
                "width": "100%",
                "willChange": "opacity",
              }
            }
          />
        </picture>
        <noscript>
          <picture>
            <source
              sizes="(min-width: 512px) 512px, 100vw"
              srcSet="/static/6efab/hexel.webp 128w,/static/20b45/hexel.webp 256w,/static/5999d/hexel.webp 512w"
              type="image/webp"
            />
            <img
              alt="alt"
              data-gatsby-image-ssr=""
              data-main-image=""
              decoding="async"
              loading="eager"
              sizes="(min-width: 512px) 512px, 100vw"
              src="/static/e82c1/hexel.jpg"
              srcSet="/static/8710f/hexel.jpg 128w,/static/0ee4a/hexel.jpg 256w,/static/e82c1/hexel.jpg 512w"
              style={
                {
                  "backgroundColor": undefined,
                  "height": "100%",
                  "left": 0,
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "opacity": 0,
                  "position": "absolute",
                  "top": 0,
                  "transform": "translateZ(0)",
                  "transition": "opacity 250ms linear",
                  "width": "100%",
                  "willChange": "opacity",
                }
              }
            />
          </picture>
        </noscript>
        <script
          dangerouslySetInnerHTML={
            {
              "__html": "const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1,e.parentNode.parentNode.querySelector("[data-placeholder-image]").style.opacity=0)}}",
            }
          }
          type="module"
        />
      </div>
    </a>
  </div>
  <div
    className="card-body"
  >
    <h3
      className="card-title"
    >
      <a
        href="/projects/test"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        name
      </a>
    </h3>
    <div
      className="card-text"
    >
      <p>
        short
      </p>
      <p>
        <b>
          Position
        </b>
        : 
        pos
      </p>
    </div>
  </div>
</div>
`;
